packages/es-components/src/components/controls/buttons/Button.js

```diff
@@ -1,107 +1,148 @@
+    background-color: ${props => props.colors.hoverBgColor};
+    border-color: ${props => props.colors.hoverBorderColor};
+    box-shadow: 0 1px 1px rgba(0, 0, 0, 0.075),
+      0 0 0 0.2rem ${props => props.colors.focusBoxShadowColor};
+    outline: 0;
   }
 
-  &:focus,
-  &:focus-within {
-    box-shadow: 0 2px 4px 4px ${props => props.theme.colors.inputFocus};
+  &:active,
+  &.pressed {
+    color: ${props => props.colors.activeTextColor};
+    background-color: ${props => props.colors.activeBgColor};
+    border-color: ${props => props.colors.activeBorderColor};
+    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.25);
   }
 
-  &:active {
-    box-shadow: none;
+  &.pressed {
+    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.25);
   }
 
-  &&:active .es-button__display {
-    box-shadow: 0 0 0 0 transparent;
-    top: 4px;
+  &:active:focus {
+    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.25),
+      0 0 0 0.2rem ${props => props.colors.focusBoxShadowColor};
   }
 
-  &[disabled],
-  &[disabled] .es-button__display {
+  &[disabled] {
+    color: #ccc;
     cursor: not-allowed;
-    opacity: 0.65;
+    background-color: #e6e6e6;
+    border-color: #e6e6e6;
+
+    &:hover {
+      background-color: #e6e6e6;
+      border-color: #e6e6e6;
+      color: #ccc;
+    }
 
     > * {
       pointer-events: none;
     }
   }
-
-  &[disabled]:hover .es-button__display {
-    background-color: ${props => props.variant.bgColor};
-  }
 `;
 
 const Button = React.forwardRef(function Button(props, ref) {
   const { children, styleType, size, block, ...other } = props;
   const theme = useTheme();
   const buttonSize = theme.buttonStyles.button.size[size];
   const variant = theme.buttonStyles.button.variant[styleType];
+  const isInheritedStyle = styleType === 'inherited';
+
+  const focusBoxShadowColor = tinycolor.mix(
+    variant.bgColor,
+    theme.colors.black,
+    14
+  );
+  focusBoxShadowColor.setAlpha(0.5);
+
+  const buttonColors = {
+    bgColor: isInheritedStyle ? 'inherited' : variant.bgColor,
+    textColor: isInheritedStyle ? 'inherited' : getTextColor(variant.bgColor),
+    borderColor: isInheritedStyle ? 'inherited' : variant.bgColor,
+    hoverBgColor: isInheritedStyle ? 'inherited' : darken(variant.bgColor, 7.5),
+    hoverBorderColor: isInheritedStyle
+      ? 'inherited'
+      : darken(variant.bgColor, 9.9),
+    focusBoxShadowColor: isInheritedStyle
+      ? theme.colors.gray4
+      : focusBoxShadowColor.toRgbString()
+  };
+  buttonColors.activeBgColor = isInheritedStyle
+    ? 'inherited'
+    : darken(buttonColors.hoverBgColor, 2.5);
+  buttonColors.activeTextColor = isInheritedStyle
+    ? 'inherited'
+    : getTextColor(buttonColors.activeBgColor);
+  buttonColors.activeBorderColor = isInheritedStyle
+    ? 'inherited'
+    : darken(buttonColors.hoverBgColor, 5);
+  buttonColors.hoverTextColor = isInheritedStyle
+    ? 'inherited'
+    : getTextColor(buttonColors.hoverBgColor);
```

Comment:
I wonder if it would be better to just have separate objects for `inherited` and all the variants, and then just have a single ternary here. Something like:
```es6
const buttonColors = isInheritedStyle ? inheritedButtonColors : variantButtonColors;
```

Reply:
I started out that way, haha. I think my original way was more lines of code so I did this instead. I'll give it another go, because I don't love this either.

Reply:
I think my main concern is not as much lines of code as it is ease of reading. 😄

Reply:
It's now much easier to read.